src/layers/microscopy-viewer-layer/microscopy-viewer-layer.js

```diff
@@ -32,44 +33,62 @@ export class MicroscopyViewerLayer extends CompositeLayer {
     ) {
       /* eslint-disable no-bitwise */
       if (this.props.useTiff) {
-        getTiffConnections({ ...this.props }).then(connections => {
-          const firstFullImage = Object.values(connections[0])[0][0]
-            .fileDirectory;
-          this.setState({
-            connections,
-            pool: new Pool(),
-            imageWidth: firstFullImage.ImageWidth,
-            imageHeight: firstFullImage.ImageLength,
-            tileSize: firstFullImage.TileWidth,
-            isTiff: true,
-            isZarr: false
-          });
-        });
+        initTiff({ ...this.props }).then(
+          ({ connections, minZoom, imageWidth, imageHeight, tileSize }) => {
+            this.setState({
+              connections,
+              minZoom,
+              imageWidth,
+              imageHeight,
+              tileSize,
+              pool: new Pool(),
+              isZarr: false,
+              isTiff: true
+            });
+          }
+        );
       } else {
-        getZarrConnections({ ...this.props }).then(connections => {
-          this.setState({ connections, isZarr: true, isTiff: false });
-        });
+        initZarr({ ...this.props }).then(
+          ({ connections, minZoom, imageWidth, imageHeight, tileSize }) => {
+            this.setState({
+              connections,
+              minZoom,
+              imageWidth,
+              imageHeight,
+              tileSize,
+              isZarr: true,
+              isTiff: false
+            });
+          }
+        );
       }
     }
   }
 
   renderLayers() {
-    const { connections, pool, imageWidth, imageHeight, tileSize } = this.state;
-    if (
-      (this.props.imageWidth && imageWidth) ||
-      (this.props.imageHeight && imageHeight) ||
-      (this.props.tileSize && tileSize)
-    ) {
-      throw new Error('If using tiff, do not set image size');
-    }
+    const {
+      connections,
+      pool,
+      imageWidth,
+      imageHeight,
+      tileSize,
+      minZoom
+    } = this.state;
+    // if (
+    //   (this.props.imageWidth && imageWidth) ||
+    //   (this.props.imageHeight && imageHeight) ||
+    //   (this.props.tileSize && tileSize)
+    // ) {
+    //   throw new Error('If using tiff, do not set image size');
+    // }
```

Comment:
we can remove this now that it is resolved

Reply:
yup sorry ...